web/secondary_toolbar.js

```diff
@@ -29,173 +29,228 @@
 var SCROLLBAR_PADDING = uiUtils.SCROLLBAR_PADDING;
 var mozL10n = uiUtils.mozL10n;
 
-var SecondaryToolbar = {
-  opened: false,
-  previousContainerHeight: null,
-  newContainerHeight: null,
+/**
+ * @typedef {Object} SecondaryToolbarOptions
+ * @property {HTMLDivElement} toolbar - Container for the secondary toolbar.
+ * @property {HTMLButtonElement} toggleButton - Button for toggling the
+ *   visibility of the secondary toolbar.
+ * @property {HTMLButtonElement} presentationModeButton - Button for toggling
+ *   presentation mode.
+ * @property {HTMLButtonElement} openFile - Button for opening a file.
+ * @property {HTMLButtonElement} print - Button for printing the document.
+ * @property {HTMLButtonElement} download - Button for downloading the document.
+ * @property {HTMLLinkElement} viewBookmark - Button for obtaining a bookmark
+ *   link to the current location in the document.
+ * @property {HTMLButtonElement} firstPage - Button for going to the first page
+ *   in the document.
+ * @property {HTMLButtonElement} lastPage - Button for going to the last page
+ *   in the document.
+ * @property {HTMLButtonElement} pageRotateCw - Button for rotating the page
+ *   clockwise.
+ * @property {HTMLButtonElement} pageRotateCcw - Button for rotating the page
+ *   counterclockwise.
+ * @property {HTMLButtonElement} toggleHandTool - Button for toggling the hand
+ *   tool.
+ * @property {HTMLButtonElement} documentPropertiesButton - Button for toggling
+ *   the visibility of a document properties overlay.
+ */
 
-  initialize: function secondaryToolbarInitialize(options, eventBus) {
-    this.eventBus = eventBus;
+/**
+ * @class
+ */
+var SecondaryToolbar = (function SecondaryToolbarClosure() {
+  /**
+   * @constructs SecondaryToolbar
+   * @param {SecondaryToolbarOptions} options
+   * @param {EventBus} eventBus
+   */
+  function SecondaryToolbar(options, eventBus) {
     this.toolbar = options.toolbar;
-    this.buttonContainer = this.toolbar.firstElementChild;
-
-    // Define the toolbar buttons.
     this.toggleButton = options.toggleButton;
     this.presentationModeButton = options.presentationModeButton;
```

Comment:
Thanks for doing this!
However, looking at the updated patch, the "Button" suffix everywhere makes the code look a bit busy. So, do we actually need it, or could we remove it for all SecondaryToolbar buttons?

I'm really sorry about not suggesting this previously, so If you do not want to remove the "...Button" part on all strings in this patch, I will *not* insist on this change!